Gcosmin

GherasimCosmin opened the pull request.

cogherasim committed:
Added gitignore
added .idea gitignore
[Task: 1]

cogherasim committed:
Java Basic 1
Implemented models for SkeletonObject, SkeletonRepository interfaces
Added exceptions PersonAlreadyExists, PersonNotFound and ClassNotInstantiated
[Task: day 2]

Commit:
new dev with playground

DIuliana committed:
Merge pull request #24 from stefan-matcovici/dev

new dev with playground

Commit:
updated readme with generics tasks

DIuliana committed:
Merge pull request #25 from stefan-matcovici/patch-1

updated readme with generics tasks

cogherasim committed:
Collections and Generics exercises
Made a collection, that holds and manage articles,
	and modified the classes from playground package to achieve the exercises goals.
[Task: Collections and Generics]

stefan-matcovici commented on src/main/java/training/endava/playground/generics/Collections/Exercises.java:
```diff
@@ -0,0 +1,83 @@
+package training.endava.playground.generics.Collections;
```
package name usually in lowercase

stefan-matcovici commented on src/main/java/training/endava/playground/generics/Collections/Exercises.java:
```diff
@@ -0,0 +1,83 @@
+import java.util.*;
+
+public class Exercises implements Comparator<Items> {
+    ArrayList<Items> list = new ArrayList<Items>();
```
as a best practice, you should let the type of the field to be as abstract as you can, basically, you call methods just from the List interface on this field
so, make this field reference type List

stefan-matcovici commented on src/main/java/training/endava/playground/generics/Collections/Items.java:
```diff
@@ -0,0 +1,37 @@
+
+public class Items {
+    private String title;
+    ArrayList<Integer> tags = new ArrayList<Integer>();
```
same with the List reference..

stefan-matcovici commented on src/main/java/training/endava/playground/generics/Collections/Items.java:
```diff
@@ -0,0 +1,37 @@
+public class Items {
+    private String title;
+    ArrayList<Integer> tags = new ArrayList<Integer>();
+    Random rand = new Random();
```
i won't let it with the default access modifier, make it clear it's private

stefan-matcovici commented on src/main/java/training/endava/playground/generics/Collections/Items.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.ArrayList;
+import java.util.Random;
+
+public class Items {
```
naming again..it's a single Item

stefan-matcovici commented on src/main/java/training/endava/playground/generics/types/LimitedContainer.java:
```diff
@@ -0,0 +1,20 @@
+    ArrayList<T> container = new ArrayList<T>();
+    N limit;
+
+    public LimitedContainer(Class<T> c, N nr) {
```
i can't understand what's this Class<T> c, parameter

stefan-matcovici commented on src/main/java/training/endava/playground/generics/types/LimitedContainer.java:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    public boolean add(T obj) {
+        if (this.container.size() < (Integer) this.limit) {
```
I don't think that's the way of comparing the filed of type T extends Number with an int. what happens if T is Double?
maybe smth like this: https://stackoverflow.com/questions/2683202/comparing-the-values-of-two-generic-numbers

Commit:
added optionals and functional interfaces

Commit:
Merge pull request #35 from stefan-matcovici/dev

added optionals and functional interfaces

cogherasim committed:
Streams, and a little bit of Optional
Added Company class,manipulated multiple companies with streams and created a collector
[Task: Java 8]

ioanasitaru commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
+//                        .map(y -> y.getName())
+//                        .reduce("", (a, b) -> a + b))
+//                .reduce("", (a, b) -> a + b);
+//        System.out.println(new Person(name, "email", "tel", 123456, companies.get(3)));
```
1. Please use flatMap instead of creating a stream inside a stream...
2. It is preferred to use method reference instead. Replace lambdas everywhere is applicable.
3. The idea of the task was to create a Person instance that has the name created as a concatenation, not just a String. Try the reduce method with new Person("") as identity value (constructor for the name attribute) and defining the rule of combining two person instances in the second parameter of the reduce method.

ioanasitaru commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
+//                forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getName())
+//                .forEach(System.out::println));
```
Please use flatMap instead of creating a stream inside a stream...

ioanasitaru commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .forEach(System.out::println));
```
Please use flatMap instead of creating a stream inside a stream...

ioanasitaru commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
+
+        //ex9
+//        Optional<Integer> upperBound = Optional.of(new Integer(1000));
+//        ArrayList<Company> companies2 = (ArrayList<Company>) makeCompanies(upperBound);
```
1. You don't need to cast List to ArrayList. ArrayList is an implementation of List.
2. Attributes should be of as abstract as you can, in case you want to change the implementation it has.

stefan-matcovici commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -51,17 +51,140 @@ public static void main(String argv[]) {
     }
 
-    public static ArrayList<Person> makeList() {
+    public static List<Company> makeCompanies( Optional<Integer> bound) {
```
no need to declare a parameter optional, you could for example overload this method with one that has no parameter

stefan-matcovici commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -51,17 +51,140 @@ public static void main(String argv[]) {
-    public static ArrayList<Person> makeList() {
+    public static List<Company> makeCompanies( Optional<Integer> bound) {
+        Integer upperBound = 6;
+        if(bound.isPresent()){
```
that's why you don't need an optional parameter, this code is the same as checking for nulll reference

ioanasitaru commented on src/main/java/training/endava/skeleton/Main.java:
```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .distinct()
+//                        .forEach(System.out::println));
```
The task requested to display the persons, so to process distinct on a stream of persons, not only on the phone numbers.
